x-pack/plugins/ml/public/application/services/new_job_capabilities/new_job_capabilities_service.ts

```diff
@@ -4,66 +4,31 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
 import {
   Field,
   Aggregation,
   AggId,
   FieldId,
-  NewJobCaps,
   EVENT_RATE_FIELD_ID,
-} from '../../../common/types/fields';
-import {
-  ES_FIELD_TYPES,
-  IIndexPattern,
-  IndexPatternsContract,
-} from '../../../../../../src/plugins/data/public';
-import { ml } from './ml_api_service';
-import { getIndexPatternAndSavedSearch } from '../util/index_utils';
-
-// called in the routing resolve block to initialize the
-// newJobCapsService with the currently selected index pattern
-export function loadNewJobCapabilities(
-  indexPatternId: string,
-  savedSearchId: string,
-  indexPatterns: IndexPatternsContract
-) {
-  return new Promise(async (resolve, reject) => {
-    if (indexPatternId !== undefined) {
-      // index pattern is being used
-      const indexPattern: IIndexPattern = await indexPatterns.get(indexPatternId);
-      await newJobCapsService.initializeFromIndexPattern(indexPattern);
-      resolve(newJobCapsService.newJobCaps);
-    } else if (savedSearchId !== undefined) {
-      // saved search is being used
-      // load the index pattern from the saved search
-      const { indexPattern } = await getIndexPatternAndSavedSearch(savedSearchId);
-      if (indexPattern === null) {
-        // eslint-disable-next-line no-console
-        console.error('Cannot retrieve index pattern from saved search');
-        reject();
-        return;
-      }
-      await newJobCapsService.initializeFromIndexPattern(indexPattern);
-      resolve(newJobCapsService.newJobCaps);
-    } else {
-      reject();
-    }
-  });
-}
+} from '../../../../common/types/fields';
+import { ES_FIELD_TYPES, IIndexPattern } from '../../../../../../../src/plugins/data/public';
+import { ml } from '../ml_api_service';
+import { processTextAndKeywordFields, NewJobCapabilitiesServiceBase } from './new_job_capabilities';
 
 const categoryFieldTypes = [ES_FIELD_TYPES.TEXT, ES_FIELD_TYPES.KEYWORD, ES_FIELD_TYPES.IP];
 
-class NewJobCapsService {
-  private _fields: Field[] = [];
-  private _catFields: Field[] = [];
-  private _dateFields: Field[] = [];
-  private _aggs: Aggregation[] = [];
-  private _includeEventRateField: boolean = true;
-  private _removeTextFields: boolean = true;
-
-  public get fields(): Field[] {
-    return this._fields;
+class NewJobCapsService extends NewJobCapabilitiesServiceBase {
+  private _catFields: Field[];
+  private _dateFields: Field[];
+  private _includeEventRateField: boolean;
+  private _removeTextFields: boolean;
+
+  constructor() {
```

Comment:
this constructor isn't needed, we don't need to change the way the private variables are given initial values.

Reply:
Removed in 4e2f344ac831a47b06266517d1dbcd0dbd0f3c48